packages/auth/amplify_auth_cognito/lib/src/auth_plugin_impl.dart

```diff
@@ -163,4 +173,158 @@ class _NativeAmplifyAuthCognito implements NativeAuthPlugin {
+          userPoolConfig,
+        );
+        final accessToken = await userPoolStorage.read(
+          key: userPoolKeys[LegacyCognitoUserPoolKey.accessToken],
+        );
+        final refreshToken = await userPoolStorage.read(
+          key: userPoolKeys[LegacyCognitoUserPoolKey.refreshToken],
+        );
+        final idToken = await userPoolStorage.read(
+          key: userPoolKeys[LegacyCognitoUserPoolKey.idToken],
+        );
+        if (accessToken != null && refreshToken != null && idToken != null) {
+          // TODO(Jordan-Nelson): fetch sign in method from keychain on iOS
+          final signInMethod = hostedUiConfig != null
+              ? CognitoSignInMethod.hostedUi
+              : CognitoSignInMethod.default$;
+          userPoolTokens = CognitoUserPoolTokens(
+            signInMethod: signInMethod,
+            accessToken: JsonWebToken.parse(accessToken),
+            refreshToken: refreshToken,
+            idToken: JsonWebToken.parse(idToken),
+          );
+        }
+      }
+    }
+
+    String? identityId;
+    AWSCredentials? awsCredentials;
+    final identityPoolId = identityPoolConfig?.poolId;
+    if (identityPoolId != null) {
+      final identityPoolStorage = getIdentityPoolStorage(
+        bundleId,
+        identityPoolId,
+      );
+      const identityPoolKeys = LegacyCognitoIdentityPoolKeys();
+      identityId = await identityPoolStorage.read(
+        key: identityPoolKeys[LegacyCognitoIdentityPoolKey.identityId],
+      );
+      final accessKeyId = await identityPoolStorage.read(
+        key: identityPoolKeys[LegacyCognitoIdentityPoolKey.accessKey],
+      );
+      final secretAccessKey = await identityPoolStorage.read(
+        key: identityPoolKeys[LegacyCognitoIdentityPoolKey.secretKey],
+      );
+      final sessionToken = await identityPoolStorage.read(
+        key: identityPoolKeys[LegacyCognitoIdentityPoolKey.sessionKey],
+      );
+      final expirationStr = await identityPoolStorage.read(
+        key: identityPoolKeys[LegacyCognitoIdentityPoolKey.expiration],
+      );
+      if (accessKeyId != null && secretAccessKey != null) {
+        DateTime? expiration;
+        if (expirationStr != null) {
+          final secondsSinceEpoch = double.tryParse(expirationStr)?.toInt();
+          if (secondsSinceEpoch != null) {
+            expiration = DateTime.fromMillisecondsSinceEpoch(
+              secondsSinceEpoch * 1000,
+              isUtc: true,
+            );
+          }
+        }
+        awsCredentials = AWSCredentials(
+          accessKeyId,
+          secretAccessKey,
+          sessionToken,
+          expiration,
+        );
+      }
+    }
+
+    if ((userPoolTokens ?? awsCredentials ?? identityId) != null) {
+      return CredentialStoreData(
+        userPoolTokens: userPoolTokens,
+        awsCredentials: awsCredentials,
+        identityId: identityId,
+      );
+    }
+    return null;
+  }
+
+  @override
+  Future<void> deleteLegacyCredentials({
```

Comment:
Do you think that when we release this we should document the fact that we are deleting the legacy credentials? Customers should not be accessing these directly and instead using the fetchAuthSession API, but they *can* access them directly, right?  

I dont' feel strongly, just wanted to call this out.

Reply:
I think it would be worth noting credential migration in in the change logs. I don't think we have to put a warning or callout for it.